Improve normalisePath

sisqbates opened the pull request:
- Perform URL decoding manually
- Remove dangerous sequences (.., //) while decoding

purplefox commented on src/test/java/io/vertx/ext/apex/UtilsTest.java:
```diff
@@ -50,22 +50,35 @@ public void testPathWithSpaces2() throws Exception {
+        "/ !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~",
+        Utils
+            .normalisePath("/%20%21%22%23%24%25%26%27%28%29%2A%2B%2C%2D%2E%2F%30%31%32%33%34%35%36%37%38%39%3A%3B%3C%3D%3E%3F%40%41%42%43%44%45%46%47%48%49%4A%4B%4C%4D%4E%4F%50%51%52%53%54%55%56%57%58%59%5A%5B%5C%5D%5E%5F%60%61%62%63%64%65%66%67%68%69%6A%6B%6C%6D%6E%6F%70%71%72%73%74%75%76%77%78%79%7A%7B%7C%7D%7E"));
   }
```
Ferran - 

I think there are quite a few more cases we should test, e.g. /foo//blah/ /foo//blah// etc

purplefox replied:
Looks good. I think we add some more unicode examples e.g. some chinese chars:

%E5%A9%B4%E5%84%BF%E6%9C%8D%E9%A5%B0"

sisqbates commented:
Tim,

just added some more tests. 

I also noticed that two test cases on SockJS were failing, since now the path normalisation removes 'redundant' slashes at the end of the path. I supposed this is the correct behavior, so I removed them, but any other suggestion is welcome.

purplefox commented on src/test/java/io/vertx/ext/apex/handler/SockJSHandlerTest.java:
```diff
@@ -88,12 +88,10 @@ private void testGreeting(String uri) {
 
     testNotFound("/echo/a");
     testNotFound("/echo/a.html");
-    testNotFound("/echo//");
```
I think these cases can be moved to the testGreeting method as they should now pass.

sisqbates commented:
Just added test for non latin characters and restored the SockJS tests (in the testGreeting method)

purplefox commented:
Thanks. Any chance you could rebase your branch on master and push it again? as it seems out of sync now

sisqbates commented:
Tim, just rebased the branch, so it should apply cleanly on master now

purplefox commented:
Thanks!
